hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -375,6 +376,8 @@ public void run() {
                 } else {
                     handleAuthenticationFailure(endpoint, request);
                 }
+            } catch (QueryResultSizeExceededException e) {
```

Comment:
The ClientEngineImpl should not deal with this responsibility. ClientEngineImpl deals with client invocations/responses.. it doesn't deal with maps or queries.